src/python/WMComponent/RucioInjector/Database/MySQL/GetCompletedBlocks.py

```diff
@@ -0,0 +1,114 @@
+                    dbsbuffer_location.pnn,
+                    dbsbuffer_dataset.path,
+                    dbsbuffer_dataset_subscription.site,
+                    dbsbuffer_workflow.name,
+                    dbsbuffer_block.create_time
+             FROM dbsbuffer_dataset_subscription
+             INNER JOIN dbsbuffer_dataset ON
+               dbsbuffer_dataset.id = dbsbuffer_dataset_subscription.dataset_id
+             INNER JOIN dbsbuffer_block ON
+               dbsbuffer_block.dataset_id = dbsbuffer_dataset_subscription.dataset_id
+             INNER JOIN dbsbuffer_file ON
+               dbsbuffer_file.block_id = dbsbuffer_block.id
+             INNER JOIN dbsbuffer_workflow ON
+               dbsbuffer_workflow.id = dbsbuffer_file.workflow
+             INNER JOIN dbsbuffer_location ON
+               dbsbuffer_location.id = dbsbuffer_block.location
+             WHERE dbsbuffer_dataset_subscription.delete_blocks = 1
+             AND dbsbuffer_dataset_subscription.subscribed = 1
+             AND dbsbuffer_block.status = 'Closed'
+             AND dbsbuffer_block.deleted = 0
+             GROUP BY dbsbuffer_block.blockname,
+                      dbsbuffer_location.pnn,
+                      dbsbuffer_dataset.path,
+                      dbsbuffer_dataset_subscription.site,
+                      dbsbuffer_workflow.name,
+                      dbsbuffer_block.create_time
+             HAVING COUNT(*) = SUM(dbsbuffer_workflow.completed)
+             """
+
+    def format(self, result):
+        """
+        _format_
+
+        Format the query results into the proper dictionary expected at the upper layer Python code.
+        The input should be a list of database objects each representing a line returned from the database
+        with key names matching the column names from the sql query
+        The result should be a list of dictionaries one record per line returned from the database
+        with key names mapped to the python code expected structures.
+
+        e.g.
+        [{'blockName': '/Cosmics/Tier0_REPLAY_2022-v425/RAW#aa3dbb67-dab0-4671-a550-711bdf9f4b08',
+          'workflowName': 'Repack_Run351572_StreamPhysics_Tier0_REPLAY_2022_ID220513161632_v425_220514_2038',
+          'location': 'T0_CH_CERN_Disk',
+          'site': {'T0_CH_CERN_Disk'},
+          'dataset': '/Cosmics/Tier0_REPLAY_2022-v425/RAW'},
+         {'blockName': '/NoBPTX/Tier0_REPLAY_2022-v425/RAW#6b8e95c3-656b-4302-8265-275df1195f4c',
+          'workflowName': 'Repack_Run351572_StreamPhysics_Tier0_REPLAY_2022_ID220513161632_v425_220514_2038',
+          'location': 'T0_CH_CERN_Disk',
+          'site': {'T0_CH_CERN_Disk'},
+          'dataset': '/NoBPTX/Tier0_REPLAY_2022-v425/RAW'}]
+
+        :param result: The result as returned by the mysql query execution.
+        :return:       List of dictionaries
+        """
+
+        # NOTE: We need to rename all the keys to follow the cammelCase standard. And also to comply
```

Comment:
I do not understand your comment here. The need to rename them is twofold:
* The rest of the code uses the variables the way they are stated here in this map. And I am talking not only about the new changes I added to RucioIjectorPoller but also about bits of code which were there from before - it does expect this set of variables rather than just the column names from the relational database (which BTW is `snake_case`). I really do not want to refactor this in the RucioInjectoerPooler and mess up with old pieces of code. Which on top of everything will enforce python variables not following our naming convention.  
* Using only the column names does not give you any kind of information to which object the value relates. Example follows:
    * while reading the code and you see just  `name` ... well it might relate to both `blockName` and `workflowName`. On top of that those bothe they come from different tables at the first place so even an column name overlap may be in place...

Bottom line, I see that map not useful but actually mandatory, instead. And It does simplify the code because it minimizes the code changes needed on the already existing code.

Reply:
Ahh, I understood your had renamed these only to follow the guidelines, but it's actually needed by RucioInjector that is already expecting such parameters (and to be consistent with the previous DAO), right?

Reply:
Correct